stdlib/private/SwiftPrivateLibcExtras/Subprocess.swift

```diff
@@ -264,9 +264,26 @@ public enum ProcessTerminationStatus : CustomStringConvertible {
 
 public func posixWaitpid(_ pid: pid_t) -> ProcessTerminationStatus {
   var status: CInt = 0
+#if CYGWIN
+  final class Box<T> {
+    let value: T
+
+    init(_ value: T) {
+      self.value = value
+    }
+  }
+
+  let unmanaged = Unmanaged.passRetained(Box<Int32>(status))
+  let statusPtr = __wait_status_ptr_t(__int_ptr: UnsafeMutablePointer<Int32>(unmanaged.toOpaque()))
+
+  if waitpid(pid, statusPtr, 0) < 0 {
+    preconditionFailure("waitpid() failed")
+  }
```

Comment:
I captured the signature in `/usr/include/sys/wait.h` on Cygwin.
```
/* Allow `int' and `union wait' for the status.  */
typedef union
  {
    int *__int_ptr;
    union wait *__union_wait_ptr;
  } __wait_status_ptr_t  __attribute__ ((__transparent_union__));

pid_t waitpid (pid_t __pid, __wait_status_ptr_t __status, int __options);
```
I'll be happy if someone suggests any simpler way.